compiler+runtime/src/cpp/jank/read/parse.cpp

```diff
@@ -15,6 +15,81 @@ namespace jank::read::parse
 {
   using namespace jank::runtime;
 
+  result<option<native_persistent_string>, native_persistent_string>
+  parse_character_in_base(native_persistent_string const &char_literal, int const base)
+  {
+    try
+    {
+      size_t chars_processed{};
+      auto const codepoint(std::stol(char_literal, &chars_processed, base));
+
+      /* `std::stol` will ignore any character that lies outside of
+       * the `base` digits range.
+       *
+       * For base `8`, valid digits are {0, 1, 2, 3, 4, 5, 6, 7}.
+       * For base `16`, valid digits are {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, a, b, c, d, e, f, A, B, C, D, E, F}
+       *
+       * If `std::stol` doesn't process all the
+       * characters in `char_literal`, we'll consider it as invalid.
+       *
+       * Refer: https://en.cppreference.com/w/cpp/string/basic_string/stol
+       */
+      if(chars_processed != char_literal.size())
+      {
+        return err("Invalid unicode digit");
+      }
+
+      std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
+
+      auto const converted(native_persistent_string{ converter.to_bytes(codepoint) });
```

Comment:
Here we're using `auto` and the `type`. If we need to specify the type, just specify the type.

```cpp
native_persistent_string const converted{ converter.to_bytes(codepoint) };
```